client/src/store/checkEmail/actions.ts

```diff
@@ -0,0 +1,10 @@
+import { showAlert } from "../../store/alert/actions";
+import API from "../api";
+
+export const checkEmail = (email: object) => (dispatch: Function) => {
+  return API.post("/forgot-password", email)
```

Comment:
where is actions? how would you follow the flow?
Where is PENDING, SUCCESS and ERROR?
why do you need redux?

Reply:
Instead of PENDING and SUCCESS actions I'm using redux-form's built-in 'submitting' props. It tells whether or not form is currently submitting. It will be true until the promise is resolved or rejected. And untill it's true, I'm showing a loading spinner.
When there is an error, i will use showAlert action, which will show it on the screen.